static/src/javascripts/projects/common/modules/identity/wizard.spec.js

```diff
@@ -0,0 +1,111 @@
+// @flow
+import { containerClassname, enhance, setPosition } from './wizard';
+
+// Not yet implemented in our codebase, see https://github.com/tmpvar/jsdom/issues/961
+HTMLElement.prototype.dataset = {};
+
+const steps = ['1', '2', '3', '4'];
+
+beforeEach(() => {
+    if (document.body) {
+        document.body.innerHTML = `
+            <div class="${containerClassname}">
+                <div class="manage-account-wizard__step">1</div>
+                <div class="manage-account-wizard__step">2</div> 
+                <div class="manage-account-wizard__step">3</div> 
+                <div class="manage-account-wizard__step">4</div> 
+                <div class="manage-account-wizard__controls-pager">
+                    <div class="js-manage-account-wizard__next"></div>
+                    <div class="js-manage-account-wizard__prev"></div>
+                </div>
+            </div>
+        `;
+    }
+    return enhance(document.getElementsByClassName(containerClassname)[0]);
+});
+
+test('containerClassname must be manage-account-wizard', () => {
+    expect(containerClassname).toEqual('manage-account-wizard');
+});
+
+test('data-position is set', () => {
+    const wizardEl = document.getElementsByClassName(containerClassname)[0];
+    expect(wizardEl.dataset.position).toEqual('0');
+});
+
+test('data-position is updated', () => {
+    const wizardEl = document.getElementsByClassName(containerClassname)[0];
+    return setPosition(wizardEl, 2).then(() => {
+        expect(wizardEl.dataset.position).toEqual('2');
+    });
+});
+
+test('data-position is updated when value is off-range', () => {
+    const wizardEl = document.getElementsByClassName(containerClassname)[0];
+    return setPosition(wizardEl, 99).then(() => {
+        expect(wizardEl.dataset.position).toEqual('0');
+    });
+});
+
+test('move-next works', () => {
+    const wizardEl = document.getElementsByClassName(containerClassname)[0];
+    const moveEl = document.getElementsByClassName(
+        'js-manage-account-wizard__next'
+    )[0];
+    moveEl.dispatchEvent(new Event('click'));
+    setTimeout(() => {
```

Comment:
Is there a change that this could be flaky? Would it be better to mock `dispatchEvent` instead?

Reply:
jsdom _seems_ to have a supported comprehensive events model, i'd leave it as as is unless it starts to fail as t's the most comprehensive way to test it (plus, if tha pagination buttons stop working we are seriously toast, there's no other way to move through the content)